config/config.go

```diff
@@ -56,5 +68,71 @@ func (c *Config) initConfigFromFile(path string) error {
 	if c.HaProxyPidFile == "" {
 		return errors.New("haproxy_pid_file is required")
 	}
+
+	if c.BackendTLS.CACertificatePath != "" {
+		pemData, err := os.ReadFile(c.BackendTLS.CACertificatePath)
+		if err != nil {
+			return err
+		}
+		for len(pemData) > 0 {
```

Comment:
Will this for run forever?? Should this be an if?

Reply:
A bunch of this was ruthlessly stolen from https://github.com/smallstep/crypto/blob/122554066d9a19620097ed6d80dab5dc0af821cd/pemutil/pem.go#L212-L231.

There definitely appears to be a case where it will run forever. In the original implementation, there are continue statements and a return that will break out of the loop. Since we are doing a simple check, and `if` should suffice.

Reply:
The implementation here only checks the first certificate (hopefully the root CA). Do we need to validate the whole bundle if there are intermediate certificates?

Reply:
We load certs to validate them here, because otherwise, that file is blindly loaded into haproxy. If haproxy has issues with it, it will be much harder to diagnose without having log entries for tcp-router. Sincew haproxy is going to load all certs in the file, we should validate that all certs in the file are parseable.